### PR TITLE
fix(#10500): When the tier is removed all tags are also removed.

### LINKED ISSUE #10500: When the tier is removed all tags are also removed.
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**

When we remove the tier, other tags also get removed, please see the recording.

**To Reproduce**

https://user-images.githubusercontent.com/59080942/224227830-4ff9216a-7343-47d6-a99c-ce19687fe06f.mov

**Expected behavior**

Only the tier should get removed not all other tags.

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.13.2]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #10500 
### Describe your changes :

I worked on issue #10500.

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/59080942/224303972-a6eaee17-89be-4205-bf97-83c025bc2fd2.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Review by chirag-madlani — COMMENTED** [review]
Can we add some test to check wether it's working well?

**Comment by Sachin-chaurasiya** [comment]
> Can we add some test to check wether it's working well?

Yes @chirag-madlani , working on it.

**Comment by Sachin-chaurasiya** [comment]
> Can we add some test to check wether it's working well?

@chirag-madlani , done, please review.

**Review by ShaileshParmar11 — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/EntityDetails.spec.js [hunk 1]
```diff
@@ -161,6 +163,31 @@ describe('Entity Details Page', () => {
...
+      .scrollIntoView()
+      .should('be.visible')
+      .type(entityTag);
+
+    verifyResponseStatusCode('@tagsRequest', 200);
+    verifyResponseStatusCode('@glossaryRequest', 200);
+
+    cy.get('.ant-select-item-option-content')
+      .first()
+      .should('be.visible')
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/EntityDetails.spec.js:186]: I don't think we are verifying case we have fixed in this PR?
Add a Tire
Add some tags
Remove Tire should not remove tags right?

Are we testing same here?

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/EntityDetails.spec.js:186]: @chirag-madlani , We are testing the same behavior.

- Add the tier, tag
- Remove the tie
- Check for entity tags and they should be present

**Review by chirag-madlani — DISMISSED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**